snuba/datasets/storage.py

```diff
@@ -167,7 +167,7 @@ def __init__(
         stream_loader: KafkaStreamLoader,
         query_splitters: Optional[Sequence[QuerySplitStrategy]] = None,
         mandatory_condition_checkers: Optional[Sequence[ConditionChecker]] = None,
-        replacer_processor: Optional[ReplacerProcessor] = None,
+        replacer_processor: Optional[ReplacerProcessor[Any]] = None,
```

Comment:
`ErrorsReplacer` is not a valid substitute for `ReplacerProcessor[Replacement]` because from `issubclass(Foo, Bar)` it does not follow that `issubclass(ReplacerProcessor[Foo], ReplacerProcessor[Bar])`.

The core issue is that there are two `Replacement` classes, one that only contains data relevant to `ReplacerWorker`, and a subclass that contains additional attributes needed by `ErrorsReplacer`.